src/main/java/uk/gov/pay/adminusers/utils/CountryConverter.java

```diff
@@ -0,0 +1,64 @@
+package uk.gov.pay.adminusers.utils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Charsets;
+import com.google.common.io.Resources;
+import com.google.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class CountryConverter {
+
+    public static final String COUNTRIES_FILE_PATH = "countries.json";
+
+    private static class Country {
+        private final String name;
+        private String isoCode;
+        private String type;
+        
+        Country(String name, String typeAndIsoCode) {
+            this.name = name;
+            String[] typeAndIsoCodeSplit = typeAndIsoCode.split(":");
+            this.type = typeAndIsoCodeSplit[0];
+            this.isoCode = typeAndIsoCodeSplit[1];
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public boolean isCountry() {
+            return type.equalsIgnoreCase("country");
+        }
+        
+        public String getIsoCode() {
+            return isoCode;
+        }
+    }
+    private final ObjectMapper objectMapper;
+    private final Map<String, String> countries;
+
+    @Inject
+    public CountryConverter(ObjectMapper objectMapper) throws IOException {
+        this.objectMapper = objectMapper;
+        String countries = Resources.toString(Resources.getResource(COUNTRIES_FILE_PATH), Charsets.UTF_8);
+        this.countries = createMap(countries);
+    }
+
+    private Map<String, String> createMap(String countries) throws IOException {
+        List<List<String>> allCountries = objectMapper.readValue(countries, new TypeReference<List>() {});
+        return allCountries.stream()
+                .map(a -> new Country(a.get(0), a.get(1)))
+                .filter(Country::isCountry)
+                .collect(Collectors.toMap(Country::getIsoCode, Country::getName)
+        );
+    }
```

Comment:
So forgive me, but I spent quite a while pondering if it’s possible to attain a higher state of lambda transcendence, where we don’t need the intermediary `Country` object at all.

I came up with this:

```java
private Map<String, String> createMap(String countries) throws IOException {
    List<List<String>> allCountries = objectMapper.readValue(countries, new TypeReference<List<String>>() {});
    Pattern countryColonLiteralThenCode = Pattern.compile("^country:([A-Z][A-Z])$");
    return allCountries.stream()
            .filter(country -> countryColonLiteralThenCode.matcher(country.get(1)).matches())
            .collect(toMap(country -> country.get(1).substring(country.get(1).indexOf(':')), country -> country.get(0)));
    }
```

Thoughts? I quite like that it cuts the class size by about a third but I’m not sure if it’s really much clearer.

Reply:
The reason why I had such a verbose way of dealing with things (inner class etc.) was indeed readability.
I like your solution, though! Trying to make it a bit more readable I came up with

```
     List<List<String>> allCountries = objectMapper.readValue(countries, new TypeReference<List<List<String>>>() {});
        return allCountries.stream()
                .filter(country -> country.get(1).startsWith("country:"))
                .collect(toMap(
                        country -> getIsoCode(country.get(1)),
                        country -> country.get(0)));
```
where `getIsoCode` does the substring bit. What do you think?